src/interfaces/pyo3/py_cover_crypt.rs

```diff
@@ -0,0 +1,404 @@
+use abe_policy::AccessPolicy;
+use cosmian_crypto_core::{
+    bytes_ser_de::{Deserializer, Serializable, Serializer},
+    KeyTrait,
+};
+use pyo3::{exceptions::PyException, exceptions::PyTypeError, prelude::*, types::PyType};
+
+use crate::{
+    api::CoverCrypt as CoverCryptRust,
+    interfaces::pyo3::py_abe_policy::Policy,
+    interfaces::statics::{
+        CoverCryptX25519Aes256, EncryptedHeader, MasterSecretKey as MasterSecretKeyRust,
+        PublicKey as PublicKeyRust, SymmetricKey as SymmetricKeyRust,
+        UserSecretKey as UserSecretKeyRust,
+    },
+};
+
+// Pyo3 doc on classes
+// https://pyo3.rs/v0.16.2/class.html
+
+#[pyclass]
+pub struct MasterSecretKey {
+    inner: MasterSecretKeyRust,
+}
+
+#[pymethods]
+impl MasterSecretKey {
+    /// Convert to bytes for sending/saving the key
+    pub fn to_bytes(&self) -> PyResult<Vec<u8>> {
+        match self.inner.try_to_bytes() {
+            Ok(res) => Ok(res),
+            Err(e) => Err(PyException::new_err(e.to_string())),
+        }
+    }
+
+    /// Read key from bytes
+    #[classmethod]
+    pub fn from_bytes(_cls: &PyType, key_bytes: Vec<u8>) -> PyResult<Self> {
+        match MasterSecretKeyRust::try_from_bytes(&key_bytes) {
+            Ok(key) => Ok(MasterSecretKey { inner: key }),
+            Err(e) => Err(PyException::new_err(e.to_string())),
+        }
+    }
+}
+
+#[pyclass]
+pub struct PublicKey {
+    inner: PublicKeyRust,
+}
+
+#[pymethods]
+impl PublicKey {
+    /// Convert to bytes for sending/saving the key
+    pub fn to_bytes(&self) -> PyResult<Vec<u8>> {
+        match self.inner.try_to_bytes() {
+            Ok(res) => Ok(res),
+            Err(e) => Err(PyException::new_err(e.to_string())),
+        }
+    }
+
+    /// Read key from bytes
+    #[classmethod]
+    pub fn from_bytes(_cls: &PyType, key_bytes: Vec<u8>) -> PyResult<Self> {
+        match PublicKeyRust::try_from_bytes(&key_bytes) {
```

Comment:
We may not want to take ownership here, though I don't really know what it means in python.

```suggestion
    pub fn from_bytes(_cls: &PyType, key_bytes: &[u8]) -> PyResult<Self> {
        match PublicKeyRust::try_from_bytes(key_bytes) {
```

Do this everywhere you can

Reply:
Seems not possible: TypeError: argument 'key_bytes': 'list' object cannot be converted to 'PyBytes'